<launch>
  <!-- Start rosserial -->
  <arg name="port" default="/dev/ttyACM0" />  <!-- Replace with your Arduino's serial port -->
  <arg name="baud" default="57600" />          <!-- Match with the Arduino sketch's baud rate -->
  <node pkg="rosserial_python" type="serial_node.py" name="rosserial" output="screen">
    <param name="port" value="$(arg port)" />
    <param name="baud" value="$(arg baud)" />
  </node>

  <!-- Start teleop_twist_joy -->
    <arg name="joy_config" default="ps3" />
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="config_filepath" default="$(find teleop_twist_joy)/config/$(arg joy_config).config.yaml" />
  <arg name="joy_topic" default="joy" />
  
  <node pkg="joy" type="joy_node" name="joy_node">
    <param name="dev" value="$(arg joy_dev)" />
    <param name="deadzone" value="0.3" />
    <param name="autorepeat_rate" value="20" />
    <remap from="joy" to="$(arg joy_topic)" />
  </node>

  <node pkg="teleop_twist_joy" name="teleop_twist_joy" type="teleop_node">
    <rosparam command="load" file="$(arg config_filepath)" />
    <remap from="joy" to="$(arg joy_topic)" />
  </node>

<node name="joy_av_node" pkg="av_senior" type="joy_av.py" output="screen"/>
 
</launch>

